CMakeLists.txt

```diff
@@ -19,10 +19,6 @@ set(VERSION_MAJOR 1)
 set(VERSION_MINOR 0)
 set(VERSION_PATCH 0)
 
-option(S2N_NO_PQ "Disables all Post Quantum Crypto code. You likely want this
-for older compilers or uncommon platforms." OFF)
-option(S2N_NO_PQ_ASM "Turns off the ASM for PQ Crypto even if it's available for the toolchain.
-You likely want this on older compilers." OFF)
```

Comment:
Should we keep these options and just have them be no-ops? Does CMake warn or error out if you pass an option that doesn't exist?

Reply:
in my experience, cmake just warns about unrecognized options.

```
CMake Warning:
  Manually-specified variables were not used by the project:

    S2N_FIPS
    S2N_INTERN_LIBCRYPTO
    S2N_NO_PQ


-- Build files have been written to: /home/ubuntu/aws-lc/build
```

also, there may be cases when users link against AWS-LC but still don't want PQ enabled (admittedly, i can't think of a strong use-case for this off the top of my head). why remove these options?

Reply:
> Why remove these options?

The `S2N_NO_PQ` flag has been used to skip compiling the NIST reference Kyber code in the pq-crypto directory of s2n. It's confusing to keep around a flag to skip compiling code that doesn't exist anymore. 

As far as I'm aware, the `S2N_NO_PQ` flag has only been enabled when it's been needed to keep s2n compiling on very old compiler versions that don't know about certain modern x86_64 instructions used in some of the assembly optimizations, and used to keep s2n compiling on some more obscure CPU architectures (MIPS, PowerPC, etc) that the NIST Kyber reference code doesn't compile for but that the AWS Common Runtime SDK targets. For both of these use cases (old compilers and obscure architectures), the `S2N_NO_PQ` flag is no longer needed.

We don't have equivalent flags to disable entire algorithms across the whole s2n-tls codebase (S2N_NO_RSA, S2N_NO_X25519, S2N_NO_CHACHAPOLY, etc). This PR is meant to bring PQ algorithms more in line with how all other crypto algorithms are treated in s2n, and have the necessary logic that detects if the algorithm is supported by the libcrypto that s2n is compiled against.